java/src/com/ibm/streamsx/topology/internal/core/StreamImpl.java

```diff
@@ -362,13 +363,25 @@ public void publish(String topic) {
         this.connectTo(op, false, null);
     }
     
-    public TStream<T> parallel(int width, Routing routing) {
+    public TStream<T> parallel(final int width, Routing routing) {
+        @SuppressWarnings("serial")
+        Supplier<Integer> supplier = new Supplier<Integer>() {
```

Comment:
Isn't there an issue here that Supplier captures the TStream instance, and so is not actually serializable?